tests/test_cases/test_cocotb/test_concurrency_primitives.py

```diff
@@ -165,3 +167,32 @@ async def coro(delay):
     test_start = cocotb.utils.get_sim_time(units="ns")
     await Combine(*coros)
     assert cocotb.utils.get_sim_time(units="ns") == test_start + max_delay
+
+
+@cocotb.test()
+async def test_recursive_combine(_):
```

Comment:
```suggestion
async def test_recursive_combine(_):
    """ Test using `Combine` on forked coroutines that themselves use `Combine`.
    
    This does not test passing `Combine` triggers into `Combine`. """
```
Maybe we should have another test for that.